addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -2973,6 +2974,66 @@ registry.many2one = SnippetOptionWidget.extend({
     }
 });
 
+/**
+ * Handle the save of a snippet as a template that can be reused later
+ */
+registry.SnippetSave = SnippetOptionWidget.extend({
+    xmlDependencies: ['/web_editor/static/src/xml/editor.xml'],
+
+    /**
+     * @override
+     */
+    isTopOption: function () {
+        return true;
+    },
+
+    //--------------------------------------------------------------------------
+    // Options
+    //--------------------------------------------------------------------------
+
+    /**
+     * @see this.selectClass for parameters
+     */
+    saveSnippet: function (previewMode, widgetValue, params) {
+        return new Promise(resolve => {
+            const dialog = new Dialog(this, {
+                title: _t("Save Your Block"),
+                size: 'small',
+                $content: $(qweb.render('web_editor.dialog.save_snippet', {
```

Comment:
Too bad [wutils.prompt](https://github.com/odoo-dev/odoo/blame/c85a6017d7041feacdd8bd46d7b1a4be850d18d9/addons/website/static/src/js/utils.js#L84) is not available in web_editor as @qsm-odoo todo suggested

Reply:
prompt should not be used anyway, it is damn ugly :p

Reply:
I used it in inner snipet for mass mailing channel creation through left panel, I don't think it is uglier than other stuff tbh.
Well anyway that should not be a reason not to use it if we could, true issue would then be to make it more beautiful :D

Reply:
>  don't think it is uglier than other stuff tbh

:worried: 

I think that it is the only code in odoo that I marked as "deprecated" :')

Reply:
We should probably decide something about it then, definitely better than having to rebuilt a complete prompt dialog

Reply:
It is a framework task, I could ask them again if they plan to do something about it but I guess their answer will be "maybe one day when owl is finished". But it is not a so good idea anyway, a prompt is not overridable easily, while a dialog template is.

Reply:
You won't convince me that
```javascript
const dialog = new Dialog(this, {
    title: _t("Save Your Block"),
    size: 'small',
    $content: $(qweb.render('web_editor.dialog.save_snippet', {
        currentSnippetName: _.str.sprintf(_t("Custom %s"), this.data.snippetName),
    })),
    buttons: [{
        text: _t("Save"),
        classes: 'btn-primary',
        close: true,
        click: async () => {
            const snippetName = dialog.el.querySelector('.o_we_snippet_name_input').value;
            // Do stuff with result
            resolve();
        },
    }, {
        text: _t("Discard"),
        close: true,
        click: () => resolve(),
    }],
}).open();
```
plus
```xml
<!-- Save Snippet Name option dialog -->
<t t-name="web_editor.dialog.save_snippet">
    <div>
        <div class="form-group form-row">
            <label class="col-form-label col-md-3" for="snippetName">Name</label>
            <div class="col-md-9">
                <input type="text"
                       class="form-control o_we_snippet_name_input"
                       id="snippetName"
                       t-att-value="currentSnippetName"/>
            </div>
        </div>
    </div>
</t>
```

is better than
```javascript
return wUtils.prompt({
    window_title: _t("Save Your Block"),
    input: _t("Name"),
}).then(function (result) {
    // Do stuff with result
});
```

Or I am missing something

Reply:
1. Try to override your prompt code to add a checkbox, you cannot
2. What if I have a long question and that my label is not a small "Name"... the columns won't adapt so you will have to create a custom template anyway
3. Leave the prompt without entering something, the promise is rejected, so the editor is stuck forever... unless you add code to handle the rejection.
4. Prompt uses a totally different API than the Dialog class, so you cannot use the same options and it will not trigger the same events than a Dialog... which the editor may listen for some reason.
5. ... try to read the prompt code :')

I am not saying a prompt should not be implemented, I am saying that the current one is old, does not allow overrides and does not contain enough options to make it useful.

Reply:
Actually, the answer of the framework team is harder: "You can do it yourself" :')

Reply:
No doubt it could be improved but why deprecating it when it fits some needs and allow a code of 6 lines instead of 30 (including an xml template) that redo the logic?

And for 1., we actually do it [here](https://github.com/odoo-dev/odoo/blame/b3c95fa5c4ae470b9db080a5e325275b3cdcb581/addons/website/static/src/js/menu/new_content.js#L81), so looks like it is possible.

You definitely know better the Dialog/js stuff so no doubts about your other points, but regarding my previous message that would be better to have a 6 lines way of calling a simple prompt input, which probably cover most cases.

Reply:
> Actually, the answer of the framework team is harder: "You can do it yourself" :')

It doesn't surprise me whatsoever, would've bet a few bucks on it.

Reply:
deprecating is not removing, I won't ask someone who uses it to remove it. I just won't encourage to use it, which is the goal of deprecation.

Note: for (1), I meant add a checkbox from outside, as another app extension. Ok you could extra the `init` in a method, call `_super` etc...

> that would be better to have a 6 lines way of calling a simple prompt input, which probably cover most cases

Of course. I guess that I could find the time to refactor it a minimum since the framework team is not gonna do it. In the end, I think that we should not have any prompt **use case** any more. Either the modal is complex and using a dialog will be necessary. Either it is just an input and it should be a small input appearing in the left panel, not a modal at all (like in this case, but don't have to implement that now :))